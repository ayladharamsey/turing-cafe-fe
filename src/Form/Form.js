// import React, { Component } from 'react';
// import './Form.css';

// class Form extends Component {
//     constructor() {
//         super();
//         this.state = {
//             name: '',
//             date: '',
//             time: '',
//             numGuests: 0
//         }
//     }

//     handleChange() {

//     }

//     handleSubmit() {

//     }

//     resetInputs() {

//     }

//     render() {
//         return (

//         ) 
//     }



// }

// export default Form;