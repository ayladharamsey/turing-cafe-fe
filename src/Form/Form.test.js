import React from 'react';
import { shallow } from 'enzyme'
import Form from './Form';
import { isJSXEmptyExpression } from '@babel/types';

describe('Form', () => {
    let wrapper, mockEvent;

    beforeEach(() => {
        wrapper = shallow(<Form addBooking = {jest.fn} />)
        mockEvent = {target: { name: 'name', value: 'Ayla'}}
    })

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a default state', () => {
        expect(wrapper.state()).toEqual({
            name: '',
            date: '',
            time: '',
            number: 0,
        });
    });

    it('should call handleChange upon change', () => {
        wrapper.instance().handleChange = jest.fn();
        wrapper.instance().forceUpdate();
        wrapper.find('.name').simulate('change', mockEvent);
        expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent);
    });

    it('should update state upon change', () => {
        expect(wrapper.state('name')).toEqual('');
        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state('name')).toEqual('Ayla')
    });

    it('should call handleSubmit upon click', () => {
        wrapper.instance().handleSubmit = jest.fn();
        wrapper.instance().forceUpdate();
        wrapper.find('button').simulate('click', mockEvent);
        expect(wrapper.instance().handleSubmit).toHaveBeenCalledWith(mockEvent);
    });

})