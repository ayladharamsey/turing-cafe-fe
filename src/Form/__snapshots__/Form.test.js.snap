// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<section>
  <input
    className="name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="date"
    name="date"
    onChange={[Function]}
    placeholder="Date"
    type="date"
    value=""
  />
  <input
    className="time"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
    value=""
  />
  <input
    className="guests"
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
    value={0}
  />
  <button
    onClick={[Function]}
  >
    Make Reservations
  </button>
</section>
`;
